app/android/runtime_client/src/org/xwalk/app/runtime/extension/XWalkRuntimeExtensionManager.java

```diff
@@ -152,10 +154,15 @@ private void loadExternalExtensions() {
         // If there is a native external extension, register the path. The
         // extensions under the path will be loaded automatically when the
         // native service starts.
-        String path = "/data/data/" + mActivity.getPackageName() + "/lib";
-        File libraryDir = new File(path);
-        if (libraryDir.listFiles().length > 0)
-            mNativeExtensionLoader.registerNativeExtensionsInPath(path);
+        String path = "";
+        try {
+            ApplicationInfo appInfo =
+                    mContext.getPackageManager()
+                    .getApplicationInfo(mActivity.getPackageName(), 0);
+            path = new File(appInfo.nativeLibraryDir).getParent();
+        } catch (final NameNotFoundException e) {
+        }
+        mNativeExtensionLoader.registerNativeExtensionsInPath(path);
```

Comment:
This method is not current. "nativeLibraryDir" is more straightforward, which be introduced  since 2.3.x.